storage/range.go

```diff
@@ -1584,7 +1596,21 @@ func (r *Range) FirstIndex() (uint64, error) {
 
 // Snapshot implements the raft.Storage interface.
 func (r *Range) Snapshot() (raftpb.Snapshot, error) {
-	panic("TODO(bdarnell): implement raft.Storage.snapshot")
+	r.RLock()
+	defer r.RUnlock()
+
+	data, err := gogoproto.Marshal(r.Desc)
```

Comment:
TODO added. The plan is of course to pack up all the data here, but that's lower-priority for now (not strictly necessary until we start truncating the old logs). Synchronizing Desc.Replicas and ConfState.Nodes comes first; once we have that we can start doing more tests of replication and recovery.